examples/lie_labs_api.py

```diff
@@ -0,0 +1,20 @@
+import torch
+
+import theseus.labs.lie as lie
+
+g = lie.rand(lie.SE3, 5)
+x = torch.rand(5, 6)
+
+z = g.retract(x)
+
+try:
+    y = g + x
+except RuntimeError as e:
+    print(e)
+
+y = g + lie.TangentTensor(x)
+
+torch.testing.assert_close(z._t, y._t)
+
+w = g + lie.cast(x, ltype=lie.tgt)
```

Comment:
Related to previous comment. rand has ltype first and cast has data first.

Reply:
The funny things is that what we probably want is exactly the opposite. `cast` receive type first, so that it's consistent with `typing.cast`, and `rand` has ltype after size, so that it's more consistent with `torch` (not 100% the same, because `dtype` is optional in torch, but not for us). 

The reason I put `ltype` first in `rand` was not API, but that I was having some errors due to the use of `*size`. But I think I can fix this.

Reply:
Fixed `rand` in the other PR.